File: server/src/pkg/domain/model/external_issue.go
package model

import (
	"time"

	"github.com/gofrs/uuid"
)

type ExternalIssue struct {
	ID           uuid.UUID `json:"id"`
	ProjectID    uuid.UUID `json:"projectId"`
	Provider     string    `json:"provider"` // "github", "gitlab", "jira"
	KeyOrNumber  string    `json:"key"` // "#123", "ABC-123"
	Title        string    `json:"title"`
	State        string    `json:"state"` // "open", "closed", etc.
	URL          string    `json:"url"`
	CreatedAt    time.Time `json:"createdAt"`
	UpdatedAt    time.Time `json:"updatedAt"`
}

// IssueState represents the possible states of an issue
type IssueState string

const (
	IssueStateOpen     IssueState = "open"
	IssueStateClosed   IssueState = "closed"
	IssueStatePending  IssueState = "pending"
)

// IssueResolveResult represents the result of resolving an issue reference
type IssueResolveResult struct {
	IssueID *uuid.UUID `json:"issueId,omitempty"`
	Key     string     `json:"key"`
	Title   string     `json:"title,omitempty"`
	State   string     `json:"state,omitempty"`
	URL     string     `json:"url,omitempty"`
	Status  string     `json:"status"` // "resolved" or "pending"
}

File: tracking-app/lib/models/external_issue.dart
class ExternalIssue {
  final String id;
  final String projectId;
  final String provider; // 'github', 'gitlab', 'jira'
  final String key;
  final String title;
  final String state;
  final String? url;
  final DateTime createdAt;
  final DateTime updatedAt;

  ExternalIssue({
    required this.id,
    required this.projectId,
    required this.provider,
    required this.key,
    required this.title,
    required this.state,
    this.url,
    required this.createdAt,
    required this.updatedAt,
  });

  factory ExternalIssue.fromMap(Map<String, dynamic> map) {
    return ExternalIssue(
      id: map['id'],
      projectId: map['projectId'],
      provider: map['provider'],
      key: map['key'],
      title: map['title'],
      state: map['state'],
      url: map['url'],
      createdAt: DateTime.parse(map['createdAt']),
      updatedAt: DateTime.parse(map['updatedAt']),
    );
  }

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'projectId': projectId,
      'provider': provider,
      'key': key,
      'title': title,
      'state': state,
      'url': url,
      'createdAt': createdAt.toIso8601String(),
      'updatedAt': updatedAt.toIso8601String(),
    };
  }
}

class IssueResolveResult {
  final String? issueId;
  final String key;
  final String? title;
  final String? state;
  final String? url;
  final String status; // 'resolved' or 'pending'

  IssueResolveResult({
    this.issueId,
    required this.key,
    this.title,
    this.state,
    this.url,
    required this.status,
  });

  factory IssueResolveResult.fromMap(Map<String, dynamic> map) {
    return IssueResolveResult(
      issueId: map['issueId'],
      key: map['key'],
      title: map['title'],
      state: map['state'],
      url: map['url'],
      status: map['status'],
    );
  }

  bool get isResolved => status == 'resolved';
  bool get isPending => status == 'pending';
}

class DescriptionSuggestionsResponse {
  final List<String> suggestions;

  DescriptionSuggestionsResponse({required this.suggestions});

  factory DescriptionSuggestionsResponse.fromMap(Map<String, dynamic> map) {
    return DescriptionSuggestionsResponse(
      suggestions: List<String>.from(map['suggestions'] ?? []),
    );
  }
}